ibis/expr/window.py

```diff
@@ -228,11 +230,15 @@ def combine(self, window):
                     "Expecting '{}' Window, got '{}'"
                 ).format(self.how.upper(), window.how.upper())
             )
-        mlb = self.max_lookback
+        prec = self.preceding
         kwds = dict(
-            preceding=self.preceding or window.preceding,
+            preceding=(prec
+                       if (isinstance(prec, (int, np.integer)) and prec)
+                       or (not isinstance(prec, (int, np.integer))
+                           and prec is not None)
```

Comment:
This is pretty dense, can you write it using traditional `if` statements?

Reply:
Yep, moved into private fn that can be used by `following` as well